artifactory/cli.go

```diff
@@ -1488,10 +1503,16 @@ func downloadCmd(c *cli.Context) {
 	}
 
 	configuration := createDownloadConfiguration(c)
-	downloaded, failed, logFile, err := generic.Download(downloadSpec, configuration)
-	defer logUtils.CloseLogFile(logFile)
-	err = cliutils.PrintSummaryReport(downloaded, failed, err)
-	cliutils.FailNoOp(err, downloaded, failed, isFailNoOp(c))
+	rtDetails := createArtifactoryDetailsByFlags(c, true)
+	buildConfiguration := createBuildToolConfiguration(c)
+	validateBuildParams(buildConfiguration.BuildName, buildConfiguration.BuildNumber)
+	downloadCommand := generic.NewDownloadCommand()
+	downloadCommand.SetConfiguration(configuration).SetBuildConfiguration(buildConfiguration).SetSpec(downloadSpec).SetRtDetails(rtDetails).SetDryRun(c.Bool("dry-run"))
+	err := commandutils.Exec(downloadCommand)
+	result := downloadCommand.Result()
+	defer logUtils.CloseLogFile(downloadCommand.LogFile())
```

Comment:
I think we better place this defer line right after 
```err := commandutils.Exec(downloadCommand)```
There are a few other places in this file where we should make this change.

Reply:
Done